internal/api/v1beta1/app_types.go

```diff
@@ -523,3 +543,18 @@ const (
 	// with a container exit code of 0, and the system is not going to restart any of these containers.
 	PodSucceeded PodState = "succeeded"
 )
+
+// Validates the key for annotations & labels. See: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set
+// https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set
+func (m *MetadataItem) Validate() error {
+	for key := range m.Apply {
+		ok, err := regexp.MatchString(`^([A-Za-z].{0,252}\/)?[A-Z0-9a-z][A-Z0-9a-z-_\.]{0,63}$`, key)
```

Comment:
we are good with this approach but there is also [validation](https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go) package